ena_portal_api/ena_handler.py

```diff
@@ -55,26 +66,30 @@ def post_request(self, data):
             response = requests.post(self.url, data=data, **get_default_connection_headers())
         return response
 
-    def get_study(self, study_sec_acc):
+    # Supports ENA primary and secondary study accessions
+    def get_study(self, study_acc):
         data = get_default_params()
         data['result'] = 'study'
         data['fields'] = 'study_accession,secondary_study_accession,study_description,study_name,study_title,' \
                          'center_name,broker_name,last_updated,first_public'
-        data['query'] = 'secondary_study_accession=\"{}\"'.format(study_sec_acc)
+        if study_acc[0:2] in ('ERP', 'SRP', 'DRP'):
+            data['query'] = 'study_accession=\"{}\"'.format(study_acc)
```

Comment:
It is the opposite around. ERPs, SRPs etc. are secondary  accessions.
```suggestion
            data['query'] = 'secondary_study_accession=\"{}\"'.format(study_acc)
```

Reply:
Refactored code for different bug, suggested changes were incorporated into code.